File: frontend/usuario.ts
export abstract class Usuario {
  cpf: string;
  nome: string;
  credenciais: { usuario: string, senha: string };

  constructor(cpf: string, nome: string, credenciais: { usuario: string, senha: string }) {
      if (new.target === Usuario) {
          throw new Error('Classe abstrata não pode ser instanciada');
      }
      this.cpf = cpf;
      this.nome = nome;
      this.credenciais = credenciais;
  }
}

File: src/biblioteca/Usuario.java
package biblioteca;

public abstract class Usuario {
    protected String cpf;
    protected String nome;
    protected Credenciais credenciais;

    public Usuario(String cpf, String nome, Credenciais credenciais) {
        this.cpf = cpf;
        this.nome = nome;
        this.credenciais = credenciais;
    }

    public String getCpf() {
        return cpf;
    }

    public String getNome() {
        return nome;
    }

    public Credenciais getCredenciais() {
        return credenciais;
    }

}

    
